runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3746,10 +3746,12 @@ remoteCompile(J9VMThread *vmThread, TR::Compilation *compiler, TR_ResolvedMethod
          int compilationSequenceNumber = compiler->getOptions()->writeLogFileFromServer(logFileStr);
          if (compiler->getOption(TR_JITServerFollowRemoteCompileWithLocalCompile) && compilationSequenceNumber)
             {
-            intptr_t rtn = 0;
             compiler->getOptions()->setLogFileForClientOptions(compilationSequenceNumber);
+            TR::MonitorTable::get()->readAcquireClassUnloadMonitor(compInfoPT->getCompThreadId());
             releaseVMAccess(vmThread);
```

Comment:
Compile locally without VM access? It seemed like we mostly do, based on `TR::CompilationInfoPerThreadBase::compile`. There is that option `TR_DisableNoVMAccess`. How does that work with JITServer enabled?

Reply:
Historically, all compilations were holding VM access which prevented the GC from doing any work. Then, we added the possibility of compiling without VM access, but introduced the classUnloadMonitor to make sure GC doesn't unload classes (it can still move objects). At the time we added the option `TR_DisableNoVMAccess` to revert to the old behavior. This older mode doesn't get used much. It may be the default at JITServer where GC does not do anything.
So, in order to obey the compilation WITH VMaccess semantics we probably should avoid releasing VM access and acquiring the classUnloadMonitor for the repeated local compilation when `getOption(TR_DisableNoVMAccess)` is true

Reply:
I see. I was thinking that that option was somehow incompatible with using JITServer, but instead we simply ignore it for remote compilations and do what we like, acquiring VM access only during the handling of a single message. I've changed the code to respect that option for the local compilation.